packages/task/src/browser/task-service.ts

```diff
@@ -244,15 +243,7 @@ export class TaskService implements TaskConfigurationClient {
 
     /** Returns an array of the valid task configurations which are configured in tasks.json files */
     async getConfiguredTasks(): Promise<TaskConfiguration[]> {
-        const taskConfigs = await this.taskConfigurations.getTasks();
-        let invalidTaskConfig: TaskConfiguration | undefined;
-        const validTaskConfigs = taskConfigs.filter(t => {
-            const isValid = this.isTaskConfigValid(t);
-            if (!isValid) {
```

Comment:
Looks like you moved this logic to `task-configurations.ts`.
The difference is:
before the changes this logic was executed every time we took the configured tasks
after: this logic is executed on events like `onDidRegisterTaskDefinition`
The problem is the logic is not executed when task schema is updated.
This leads to the bug, that some tasks are considered as invalid after updating task schema and are not available for running

Reply:
wdyt about changes https://github.com/eclipse-theia/theia/compare/reorganizeTasks?expand=1
I'm going to test it and create the PR